lib/Target/ExportSMTLIB/ExportSMTLIB.cpp

```diff
@@ -334,10 +339,28 @@ struct ExpressionVisitor
     newInfo.stream.indent(newInfo.indentLevel);
     if (failed(printExpression(worklist, newInfo)))
       return failure();
+
+    // llvm::outs()<<"\n\n\nEXPRESSION\n ";
+
+
+    // printExpression(worklist, newInfo);
+
     info.stream << info.valueMap.lookup(yieldedValue);
-    for (unsigned k = 0; k < newInfo.openParens + 1; ++k)
+
+    // llvm::outs()<<"\n\n\nOpen Parens is "<<newInfo.openParens;
+
+    if(weight != 0 && newInfo.openParens > 0)
+      info.stream <<") :weight "<< weight << ")";
```

Comment:
What if there is more than one `let` expression inside the quantifier? Don't we need to close all of them before the `:weight` attribute then?

Reply:
as far as I understood from the docs yes: `:weight` should be after the `let ... expr` where `expr` is the body of the quantifier, no matter how many other expressions/`let`s are nested in there

Reply:
Yes, that's correct! But what happens if there is one more `let` inside the `!`?
I.e., like this: ` (assert (let (( tmp1 ( exists ((x Int) (y Int)) ( ! ( let ((tmp2 (exists ((z Int)) (= z y)))) (let ((tmp3 (and tmp2 tmp2))) tmp3)) :weight 2)))) tmp1))`
Now there need to be two closing parens before `:weight`, but if I understand the code correctly, only one would be emitted currently.

Reply:
Oh I see what you mean now! I'll try to generate one such example and see what happens